internal/tsc_wrapped/strict_deps.ts

```diff
@@ -68,8 +71,15 @@ export function checkModuleDeps(
   function stripExt(fn: string) {
     return fn.replace(/(\.d)?\.tsx?$/, '');
   }
-  const allowedMap: {[fileName: string]: boolean} = {};
-  for (const d of allowedDeps) allowedMap[stripExt(d)] = true;
+  const allowedFiles: Set<string> = new Set();
+  const allowedDirs: Set<string> = new Set();
+  for (const d of allowedDeps) {
+    const path = stripExt(d);
```

Comment:
this assumes that a file has an extension and a directory doesn't? feels like the wrong condition.

Reply:
This assumes that a "directory" is anything that doesn't end in `.ts` or `.d.ts` or `.tsx` (stripExt should be stripTsExt IMO). I think this is OK because the rule would treat those as files anyway.